packages/providers/src/transaction-response/getDecodedLogs.ts

```diff
@@ -8,15 +8,13 @@ export function getDecodedLogs<T = unknown>(
   receipts: Array<TransactionResultReceipt>,
   abiInterface: Interface
 ): T[] {
-  return receipts.reduce((logs, r) => {
+  return receipts.reduce((logs: T[], r) => {
     if (r.type === ReceiptType.LogData) {
-      return logs.concat(...abiInterface.decodeLog(r.data, r.val1.toNumber(), r.id));
+      logs.push(abiInterface.decodeLog(r.data, r.val1.toNumber(), r.id)[0]);
```

Comment:
It's taking only the `0` element as it's only interested in the value of the log, not the offset.